### PR TITLE
Fix connection leak when using the api/util from the metadata comp

### PR DESCRIPTION
**Pull request by hush-hush**
### What does this PR do?

Fix connection leak when using the api/util from the metadata comp

### Describe how to test/QA your changes

Security-agent team to test if connection leak is gone.

### CODE REVIEW

**Review by pgimalac — APPROVED** [review]
LGTM
Eventually we should either reuse clients or use `CloseConnection`

**Review by paulcacheux — APPROVED** [review]

**Review by usamasaqib — APPROVED** [review]

File: pkg/config/settings/http/client.go [hunk 1]
```diff
@@ -21,15 +21,16 @@ type runtimeSettingsHTTPClient struct {
 	c                 *http.Client
 	baseURL           string
 	targetProcessName string
+	closeConnection   util.ShouldCloseConnection
 }
 
 // NewClient returns a client setup to interact with the standard runtime settings HTTP API
-func NewClient(c *http.Client, baseURL string, targetProcessName string) settings.Client {
-	return &runtimeSettingsHTTPClient{c, baseURL, targetProcessName}
+func NewClient(c *http.Client, baseURL string, targetProcessName string, closeConnection util.ShouldCloseConnection) settings.Client {
```
Comment (clamoriniere) [inline pkg/config/settings/http/client.go:28]: for improving extensibility WDTH about adding the new parameter in a ClientOptions struct so if never later we need to add another parameter we don't need to update all the function call. 

```golang
type ClientOptions struct {
     CloseConnection util.ShouldCloseConnection
}
```

```suggestion
func NewClient(c *http.Client, baseURL string, targetProcessName string, ClientOptions{CloseConnection: util.ShouldCloseConnection}) settings.Client {
```

**Review by kkhor-datadog — APPROVED** [review]

**Review by AliDatadog — APPROVED** [review]

**Pull request merged**